internal/provider/resource_custom_location.go

```diff
@@ -0,0 +1,219 @@
+				Optional:         true,
+				ValidateFunc:     DescriptionValidator,
+				DiffSuppressFunc: DiffSuppressDescription,
+			},
+			"tags": {
+				Type:        schema.TypeMap,
+				Description: "Key-value map of resource tags.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:     true,
+				ValidateFunc: TagValidator,
+			},
+			"cloud_provider": {
+				Type:        schema.TypeString,
+				Description: "Cloud Provider of the location.",
+				Required:    true,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Indication if location is enabled.",
+				Required:    true,
+			},
+			"ip_ranges": {
+				Type:        schema.TypeSet,
+				Description: "A list of IP ranges of the location.",
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"self_link": {
+				Type:        schema.TypeString,
+				Description: "Full link to this resource. Can be referenced by other resources.",
+				Computed:    true,
+			},
+		},
+		Importer: &schema.ResourceImporter{},
+	}
+}
+
+func resourceCustomLocationCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	c := m.(*client.Client)
+
+	// Define & Build
+	locationName := d.Get("name").(string)
+	isLocationEnabled := d.Get("enabled").(bool)
+	description := GetString(d.Get("description").(string))
+	tags := GetStringMap(d.Get("tags"))
+	provider := d.Get("cloud_provider").(string)
+
+	if provider != "byok" {
+		return diag.Errorf("provider must be byok for custom locations")
+	}
+
+	location := &client.Location{}
+	location.Name = &locationName
+	location.Spec = &client.LocationSpec{}
+	location.Spec.Enabled = GetBool(isLocationEnabled)
+	location.Provider = &provider
+	location.Tags = tags
+	location.Description = description
+	location.Status = nil
+	createdLocation, _, err := c.CreateCustomLocation(*location)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return setCustomLocationResource(d, createdLocation)
+}
+
+func resourceCustomLocationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	c := m.(*client.Client)
+
+	location, code, err := c.GetLocation(d.Id())
+
+	if code == 404 {
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return setCustomLocationResource(d, location)
+}
+
+func resourceCustomLocationUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	if d.HasChanges("tags", "enabled", "description") {
+		c := m.(*client.Client)
+
+		// Define & Build
+		locationToUpdate := client.Location{
+			Spec: &client.LocationSpec{
+				Enabled: GetBool(d.Get("enabled").(bool)),
+			},
+		}
+
+		locationToUpdate.Name = GetString(d.Get("name").(string))
+
+		if d.HasChange("tags") {
+			locationToUpdate.Tags = GetTagChanges(d)
+		}
+
+		if d.HasChange("description") {
+			locationToUpdate.Description = GetString(d.Get("description").(string))
```

Comment:
You must use `GetDescriptionString`. Look at the following example:
https://github.com/controlplane-com/terraform-provider-cpln/blob/main/internal/provider/resource_gvc.go#L241